x-pack/platform/plugins/shared/inference/server/chat_complete/anonymization/regex_worker_wrapper.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+const { workerData } = require('worker_threads');
+
+// When we run from source the worker file is still .ts → transpile on-the-fly
+if (workerData.fullpath.endsWith('.ts')) {
+  // eslint-disable-next-line import/no-extraneous-dependencies
+  require('ts-node').register({ transpileOnly: true });
+}
+
+// Compiled worker implementation (the .js file in production)
+// eslint-disable-next-line import/no-dynamic-require
+module.exports = require(workerData.fullpath);
```

Comment:
I don't understand why we need this. In dev we use `setup_node_env` and need to compile TS to JS at runtime, in prod we compile ahead of time to commonjs, and just need `setup_node_env/dist`: https://github.com/dgieselaar/kibana/blob/48168a82cd3a9f69bb553e4ee0f731c28cbb81c2/src/core/packages/worker-threads/server-internal/src/route_worker_entry.js#L12-L18

Reply:
to clarify, we don't use ts-node in dev, we use babel, and that contains some other stuff like resolving file imports, and setup_node_env also does some security things

Reply:
Thanks.  I used the environment condition and adjusted the paths slightly to find setup_node_env.